### DISCUSSION TITLE
Welcome to gopro-py-api Discussions!

### DISCUSSION THREAD

**Discussion by KonradIT** [KonradIT/gopro-py-api#158] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shivadhanushbr** [comment]
Hi KonradIT, this is a very impressive Library build for general use. 

I was wondering if we can use GoPro connected via HDMI cable with ur library. Wifi connection will block the use of internet over the system. Do have any solution for this with hdmi.

**Reply by shivadhanushbr** [reply]
Sorry i forgot to introduce myself. 
Shivadhanush here, IT engineer working for a regular company.

I am a techno enthusist trying to build a system that can stream from my GoPro 5 to Local System via HDMI and add some overlay or framelevel modifications and stream it back to other place like youtube.

**Reply by KonradIT** [reply]
Yeah I think so

**Reply by shivadhanushbr** [reply]
Can you please help with the steps how to achieve that, I ran though you library but I ended up not getting anything for that. Can you please guide to connect to gopro with hdmi cable.

**Reply by shivadhanushbr** [reply]
Apologies for late reply, I missed this reply from you here.

**Reply by KonradIT** [reply]
You need to be more descriptive with the issues you're having, what your end goals are and what have you tried before reaching out. 

>Can you please guide to connect to gopro with hdmi cable.

What? You take the HDMI cable, plug it into the GoPro, then you take the other end, and plug it onto a capture card or whatever system you have that can capture an HDMI input. 

> was wondering if we can use GoPro connected via HDMI cable with ur library

Yes. 

>Wifi connection will block the use of internet over the system. Do have any solution for this with hdmi.

GoPro cameras send video signals over HDMI, you cannot control the camera over HDMI...

You have to connect to the camera's WiFi *while* it's connected via HDMI (I have a Hero5 too, and I have actually connected to the WiFi and switched modes while the camera was connected to HDMI and outputting video).

**Reply by shivadhanushbr** [reply]
Apologies that I was just looking for the solution without explaining the problem statement.

I had been on building an desktop application with python which collects the video from gopro and stream its content to the application UI, and I want to give controls to user that whenever user want he/she can start recording a video out of the stream and stop as well. (Start recording, stop recording option with that application)
I don't want to give users much dependency on hardware device to use along with this application, or other application to be used with this. So single application should be used by user without any dependency on 3rd hardware/software.
But reading gopro without video was a issues, so was wondering if your library can give any solution for this.

But after your comments I understood that, we can do this without video card, right Konradit?

I just want the stream from gopro with any hardware or other service, purely only using python and capture the video and stream to the application Ui for user.

**Reply by KonradIT** [reply]
Yes. You can get the video feed from the GoPro over WiFi.

**Reply by shivadhanushbr** [reply]
But that will block the user from using the WiFi for Internet, just wanted to avoid this so I was sticking to HDMI inputs.

**Reply by KonradIT** [reply]
The user could use Ethernet to get internet. There's really no solution for the Hero5 other than to use HDMI if you depend on the Network Card to be unavailable to use.

**Reply by shivadhanushbr** [reply]
Got it sir, that looks to be an forceful alternative for user to use Ethernet. I will try to continue to work on this. Thanks a lot for your time so far.

**Comment by kilian2323** [comment]
Hi KonradIT! I am Kilian, a hobbyist Python programmer from Germany. (Usually working with Java, but also C++).
I am using the basic functions of this fantastic library with great pleasure. But I would like to dive deeper, and for that I am missing some extensive documentation. I want to control a GoPro 7 BLACK via WIFI, without the detour of connecting to it through Bluetooth first (SSID and PW are known and fixed anyway).

Now, the main things already work: connecting, starting and stopping the recording, downloading the last media file.

But when I want to try functions like "clipStatus(self, status)", I don't know what the argument "status" should look like. Can you help me? Is there a place where I can find a more extensive documentation of this API? Maybe even with examples for every function?
